File: 1961/1961.py
def main() -> None:
    p, n = (int(x) for x in input().strip().split(' '))
    altura_canos = [int(x) for x in input().strip().split(' ')]
    
    for i in range(1, n):
        cano_atual = altura_canos[i-1]
        cano_a_pular = altura_canos[i]
        diff = abs(cano_atual - cano_a_pular)
        if diff > p:
            print("GAME OVER")
            return
        
    print("YOU WIN")

if __name__ == "__main__":
    main()

File: 1961/1961.rs
use std::io;

fn input() -> String {
    let mut entrada : String = String::new();
    io::stdin()
        .read_line(&mut entrada)
        .expect("Erro ao ler a entrada!");
    entrada.to_string()
}

fn main(){
    let entrada : String = input();
    let valores : Vec<usize> = entrada
        .trim()
        .split_whitespace()
        .map(|s| s.parse().expect("msg"))
        .collect();

    let p: u32 = valores[0] as u32;
    let n: usize = valores[1];

    let altura_canos : String = input();
    let alturas : Vec<u32> = altura_canos
        .split_whitespace()
        .take(n)
        .map(|s| s.parse::<u32>().expect("msg"))
        .collect();

    for i in 1..n{
        let cano_atual : u32 = alturas[i-1]; 
        let cano_a_pular : u32 = alturas[i];

        let diff : u32 = if cano_atual > cano_a_pular {
            cano_atual - cano_a_pular
        } else {
            cano_a_pular - cano_atual
        };

        if diff > p{
            println!("GAME OVER");
            return;
        }
    }
    
    println!("YOU WIN")
}